### PR TITLE
fix(redis-4): omit credentials from db.connection_string span attribute

### PR DESCRIPTION
**Pull request by jarlefosen**
## Which problem is this PR solving?

* Omit credentials from being exposed on redis-4 span attribute `db.connection_string`

## Short description of the changes

Before attaching `db.connection_string` attribute to spans created by redis-4 we will remove credentials from the connection URL.
Specifically we remove Basic Auth and the query parameter `user_pwd` and reconstruct the URL to provide as much context as possible without leaking credentials.

### CODE REVIEW

**Review by haddasbronfman — APPROVED** [review]
thanks for adding this :)

**Review by martinkuba — APPROVED** [review]

**Review by blumamir — COMMENTED** [review]
Thanks for submitting this PR!

LGTM, Added 2 minor comments before this can merge

**Comment by haddasbronfman** [comment]
@jarlefosen Hi, do you want to keep working on that? There are only a few small fixes left.

**Comment by jarlefosen** [comment]
Hi @haddasbronfman, I sure do intend to update the PR according to comments. But at the moment I'm on vacation and will be updating as soon as I'm near my computer.
Terrible timing, sorry for delaying this.

**Comment by haddasbronfman** [comment]
@jarlefosen No rush. just wanted to know if you want to proceed with this. thanks!

File: plugins/node/opentelemetry-instrumentation-redis-4/src/utils.ts [hunk 1]
```diff
@@ -23,6 +23,28 @@ export function getClientAttributes(options: any) {
...
+function removeCredentialsFromDBConnectionStringAttribute(url?: unknown): string | undefined {
+  if (typeof url !== "string" ) {
+    return
+  }
+
+  try {
+    const u = new URL(url);
+    u.searchParams.delete("user_pwd")
+    u.username = ""
+    u.password = ""
```
[the remark this answers is not in the document]
Comment (jarlefosen) [reply plugins/node/opentelemetry-instrumentation-redis-4/src/utils.ts]: Terribly sorry about this, I have now updated the code style

File: plugins/node/opentelemetry-instrumentation-redis-4/src/utils.ts [hunk 2]
```diff
@@ -23,6 +23,28 @@ export function getClientAttributes(options: any) {
...
+    return u.href
+  } catch (e) {}
```
[the remark this answers is not in the document]
Comment (jarlefosen) [reply plugins/node/opentelemetry-instrumentation-redis-4/src/utils.ts]: Diag is now being passed as arguments and errors are logged 👍

**Review by blumamir — APPROVED** [review]
Thanks 💯

**Pull request merged**